include/Surelog/Design/DesignComponent.h

```diff
@@ -78,7 +76,7 @@ class DesignComponent : public ValuedComponentI, public PortNetHolder {
   virtual unsigned int getSize() const = 0;
   virtual VObjectType getType() const = 0;
   virtual bool isInstance() const = 0;
-  virtual const std::string& getName() const = 0;
+  virtual std::string getName() const = 0;
```

Comment:
Wouldn't we want to strive for `std::string_view` instead ? You've now worsened the situation by changing a reference to a copy.